metric-anomaly-task-manager/src/main/java/org/hypertrace/alert/engine/metric/anomaly/task/manager/job/DbRuleSource.java

```diff
@@ -0,0 +1,161 @@
+    return requestContext
+        .call(
+            () ->
+                notificationRuleStub.getAllNotificationRules(
+                    GetAllNotificationRulesRequest.newBuilder().build()))
+        .getNotificationRulesList();
+  }
+
+  private Map<String, JsonNode> getMetricAnomalyJsonMap(RequestContext requestContext) {
+    return requestContext
+        .call(
+            () ->
+                this.configServiceBlockingStub.getAllConfigs(
+                    GetAllConfigsRequest.newBuilder()
+                        .setResourceName(this.resourceName)
+                        .setResourceNamespace(this.resourceNamespace)
+                        .build()))
+        .getContextSpecificConfigsList()
+        .stream()
+        .map(ContextSpecificConfig::getConfig)
+        .map(this::convert)
+        .filter(v -> v.getRight() != null)
+        .collect(Collectors.toMap(Pair::getLeft, Pair::getRight));
+  }
+
+  private List<Document> mergeNotificationRuleEventCondition(
+      List<NotificationRule> notificationRules, Map<String, JsonNode> idMetricAnomalyJsonMap) {
+    return notificationRules.stream()
+        .filter(
+            rule ->
+                idMetricAnomalyJsonMap.containsKey(
+                    rule.getNotificationRuleMutableData().getEventConditionId()))
+        .map(
+            rule -> {
+              try {
+                JsonNode metricAnomalyNode =
+                    idMetricAnomalyJsonMap.get(
+                        rule.getNotificationRuleMutableData().getEventConditionId());
+                JsonNode ruleNode =
+                    OBJECT_MAPPER
+                        .readTree(JsonFormat.printer().print(rule))
+                        .get(NOTIFICATION_RULE_MUTABLE_DATA_KEY);
+                ((ObjectNode) ruleNode).set(AlertTaskConverter.EVENT_CONDITION, metricAnomalyNode);
+                return new JSONDocument(ruleNode);
+              } catch (IOException e) {
+                throw new UncheckedIOException(e);
+              }
+            })
+        .collect(Collectors.toList());
+  }
+
+  private Pair<String, JsonNode> convert(Value value) {
+    JsonNode jsonNode;
+    try {
+      jsonNode = OBJECT_MAPPER.readTree(JsonFormat.printer().print(value));
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+    if (jsonNode.get(ID) == null) {
+      throw new RuntimeException(String.format("id is missing in the object %s", value));
+    }
+    String id = jsonNode.get(ID).textValue();
+    if (jsonNode.get(EVENT_CONDITION_DATA_KEY) != null) {
+      jsonNode = jsonNode.get(EVENT_CONDITION_DATA_KEY);
+    } else if (jsonNode.get(EVENT_CONDITION_MUTABLE_DATA_KEY) != null) {
+      jsonNode = jsonNode.get(EVENT_CONDITION_MUTABLE_DATA_KEY);
+    } else {
+      throw new RuntimeException(
+          String.format("Event condition is missing in the object %s", value));
+    }
+    return Pair.of(id, jsonNode.get(METRIC_ANOMALY_DATA_KEY));
```

Comment:
This - `jsonNode.get(METRIC_ANOMALY_DATA_KEY)` - can be null. So, can we just check here only if the EventCondition of type MetricAnomalyEventCondtion, then only we will add in the map, else we will skip such event?